sqlite_query.c

```diff
@@ -648,12 +741,121 @@ sqlite_bind_sql_var(Form_pg_attribute att, int attnum, Datum value, sqlite3_stmt
 					char	   *outputString = NULL;
 					Oid			outputFunctionId = InvalidOid;
 					bool		typeVarLength = false;
+
 					getTypeOutputInfo(type, &outputFunctionId, &typeVarLength);
 					outputString = OidOutputFunctionCall(outputFunctionId, value); /* uuid text belongs to ASCII subset, no need to translate encoding */
 					ret = sqlite3_bind_text(stmt, attnum, outputString, -1, SQLITE_TRANSIENT);
 				}
 				break;
 			}
+		case MACADDROID:
+			{
+				int	sqlite_aff = SQLITE_NULL;
+				if (relid)
+				{
+					char * optv = get_column_option_string (relid, attnum, "column_type");
+
+					elog(DEBUG3, "sqlite_fdw : column_type affinity %s ", optv);
+					sqlite_aff = sqlite_affinity_code(optv);
+				}
+				if (sqlite_aff == SQLITE_BLOB)
+				{
+					unsigned char  *mca = palloc0(MACADDR_LEN);
+					macaddr		   *m = DatumGetMacaddrP(value);
+
+					elog(DEBUG2, "sqlite_fdw : bind mac as blob");
+					mca[0] = m->a;
+					mca[1] = m->b;
+					mca[2] = m->c;
+					mca[3] = m->d;
+					mca[4] = m->e;
+					mca[5] = m->f;
+					ret = sqlite3_bind_blob(stmt, attnum, mca, MACADDR_LEN, SQLITE_TRANSIENT);
+				}
+				else if (sqlite_aff == SQLITE3_TEXT)
+				{
+					char	   *outputString = NULL;
+					Oid			outputFunctionId = InvalidOid;
+					bool		typeVarLength = false;
+					elog(DEBUG2, "sqlite_fdw : bind mac as text");
+					getTypeOutputInfo(type, &outputFunctionId, &typeVarLength);
+					outputString = OidOutputFunctionCall(outputFunctionId, value); /* MAC text belongs to ASCII subset, no need to translate encoding */
+					ret = sqlite3_bind_text(stmt, attnum, outputString, -1, SQLITE_TRANSIENT);
+				}
+				else
+				{
+					/* MAC as integer, default */
+					macaddr *m = DatumGetMacaddrP(value);
+					sqlite3_uint64 dat = 0;
+
+					elog(DEBUG2, "sqlite_fdw : bind mac as integer");
+					dat = ((sqlite3_int64)(m->a) << (CHAR_BIT*5))
+						+ ((sqlite3_int64)(m->b) << (CHAR_BIT*4))
+						+ ((sqlite3_int64)(m->c) << (CHAR_BIT*3))
+						+ ((sqlite3_int64)(m->d) << (CHAR_BIT*2))
+						+ ((sqlite3_int64)(m->e) << (CHAR_BIT*1))
+						+ ((sqlite3_int64)(m->f) << (CHAR_BIT*0));
+					elog(DEBUG2, "sqlite_fdw : bind mac68 as integer %lld", dat);
+					ret = sqlite3_bind_int64(stmt, attnum, dat);
+				}
+				break;
+			}
+		case MACADDR8OID:
```

Comment:
Tô many duplicated code. Please refactor them.

Reply:
Refactored, please confirm current implementation.

Reply:
It is better now. Thanks. Please help me to resolve this comment.